### PR TITLE
[docs] CLI updates

### PR DESCRIPTION
**Pull request by jrhodie**

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/cli/cli.cpp [hunk 1]
```diff
@@ -7192,6 +7204,26 @@ template <> otError Interpreter::Process<Cmd("sntp")>(Arg aArgs[])
...
+     * @code
+     * sntp query 64:ff9b::d8ef:2308
+     * SNTP response - Unix time: 1540898611 (era: 0)
+     * Done
+     * @endcode
+     * @cparam sntp query [@ca{SNTP server IP}] [@ca{SNTP server port}]
+     * @par
+     * Sends an SNTP query to obtain the current unix epoch time (from January 1, 1970).
+     * - SNTP server default IP address: 2001:4860:4806:8:: (Google IPv6 NTP Server)
+     * - SNTP server default port: 123
```
Comment (Vyrastas) [inline src/cli/cli.cpp:7223]: port value should be code font

`123`

File: src/cli/cli.cpp [hunk 2]
```diff
@@ -7294,6 +7326,35 @@ template <> otError Interpreter::Process<Cmd("state")>(Arg aArgs[])
...
+     * child
+     * Done
+     * @endcode
+     * @code
+     * state leader
+     * Done
+     * @endcode
+     * @cparam state [@ca{child} | @ca{router} | @ca{leader} | @ca{detached}]
+     * @par
+     * Returns the current role of the Thread device, or changes the role as specified with one of the  options.
```
Comment (Vyrastas) [inline src/cli/cli.cpp:7342]: double space between the and options

File: src/cli/cli.cpp [hunk 3]
```diff
@@ -7294,6 +7326,35 @@ template <> otError Interpreter::Process<Cmd("state")>(Arg aArgs[])
...
+     * Possible values returned when inquiring about the device role:
+     * - `child`: The device is currently operating as a Thread child.
+     * - `router`: The device is currently operating as a Thread router.
+     * - `leader`: The device is currently operating as a Thread leader.
+     * - `detached`: The device is not currently participating in a Thread network/partition.
+     * - `disabled`: The Thread stack is currently disabled.
+     * @par
+     * Using one of the options allows you to change the current role of a device, with the excluson of
+     * changing to or from `disabled` state.
```
Comment (Vyrastas) [inline src/cli/cli.cpp:7351]: to or from a `disabled` state

**Review by Vyrastas — CHANGES_REQUESTED** [review]
Looks really good, just a few things

File: src/cli/cli.cpp (hunk 2 above)
Comment (Vyrastas) [reply src/cli/cli.cpp:7342]: Double space still there

File: src/cli/cli.cpp [hunk 4]
```diff
@@ -7294,6 +7326,35 @@ template <> otError Interpreter::Process<Cmd("state")>(Arg aArgs[])
...
+     * @par
+     * Returns the current role of the Thread device, or changes the role as specified with one of the  options.
+     * Possible values returned when inquiring about the device role:
+     * - `child`: The device is currently operating as a Thread child.
+     * - `router`: The device is currently operating as a Thread router.
+     * - `leader`: The device is currently operating as a Thread leader.
+     * - `detached`: The device is not currently participating in a Thread network/partition.
+     * - `disabled`: The Thread stack is currently disabled.
+     * @par
+     * Using one of the options allows you to change the current role of a device, with the excluson of
```
Comment (Vyrastas) [inline src/cli/cli.cpp:7350]: excluson --> exclusion

**Review by Vyrastas — CHANGES_REQUESTED** [review]

File: src/cli/cli.cpp (hunk 2 above)
Comment (Vyrastas) [reply src/cli/cli.cpp:7342]: Double space still there!

`the  options` --> `the options`

**Review by Vyrastas — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**